File: Python/통계학_.py
import sys
input = sys.stdin.readline

def get_mode(nums):
  counts = {}
  max_count = 0
  for num in nums:
    if(not counts.get(num)):
      counts[num] = 1
    else:
      counts[num] += 1
    
    max_count = max(max_count, counts[num])

  modes = []
  for num, count in counts.items():
    if(count == max_count):
      modes.append(num)

  if(len(modes) == 1):
    res = modes[0]
  else:       
    modes.sort()
    res = modes[1]
  return res

if __name__ == '__main__':
  N = int(input())
  nums = [int(input()) for _ in range(N)]

  nums.sort()
  print(round(sum(nums)/N))
  print(nums[N//2])
  print(get_mode(nums))
  print(max(nums) - min(nums))

File: C++/통계학.cpp
#include <iostream>
#include <algorithm>
#include <cmath>
#include <vector>
using namespace std;

int main()
{
	ios::sync_with_stdio(0); cin.tie(0); cout.tie(0);

	int N;
	cin >> N;

	int counts[8001] = {0,};
	vector<int> nums;
	int a = 0;
	int maxCount = 0;
	for (int i = 0; i < N; i++) {
		int tmp;
		cin >> tmp;
		nums.push_back(tmp);

		a += tmp;
		counts[tmp+4000]++;
		if (maxCount < counts[tmp + 4000]) maxCount = counts[tmp + 4000];
	}

	cout << round((double)a / (double)N) << "\n";

	sort(nums.begin(), nums.end());
	cout << nums[N/2] << "\n";

	int c;
	int cnt = 0;
	for (int i = 0; i < 8001; i++) {
		if (counts[i] == maxCount) {
			c = i-4000;
			cnt++;
			if (cnt == 2) break;
		}
	}
	cout << c << "\n";

	cout << nums[N - 1] - nums[0] << "\n";
}
